### ISSUE TITLE
Allow disabling Sankey sorting

### ISSUE THREAD

**Issue by daformat** [recharts/recharts#3683]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
When using Sankey charts to create cashflow charts, we want to group expenses or income by category. The automatic sorting makes chart links overlap and cross each other, when we want to actually group things by topic to keep the cashflow analysis legible. The fact that one expense in one category is smaller than others in other categories shouldn't move the link around and mix it with other categories, as it makes the chart more complicated to read. The goal is to give control to the user on the grouping of the branch, while still providing with the option to sort data by default

### What does the proposed API look like?
I propose to add a boolean `sort` prop to the Sankey component, with a default value of true (to no break existing implementation), so the user can control how their chart will behave.
I've prepared the [code changes](https://github.com/recharts/recharts/compare/master...daformat:recharts:feat/allow-ignoring-sankey-sorting) for that, ready to submit a PR.

**Comment by daformat** [comment]
Here is a visual example with the current sorting, and one without, to better explain the use-case:
With sorting | Without sorting
--|--
<img width="813" alt="image" src="https://github.com/recharts/recharts/assets/1694893/db511197-d045-4252-b83a-f093afd1bc3e"> | <img width="810" alt="image" src="https://github.com/recharts/recharts/assets/1694893/5941b06a-d2ec-460e-9bcd-47a20ccf4cc2">

**Comment by daformat** [comment]
It also seems that the sorting doesn't really sort the data, at least not the way one would expect, as you can see on this example made via the storybook:
<img width="1135" alt="image" src="https://github.com/recharts/recharts/assets/1694893/6b6459fe-6ed0-40a3-9c93-90724d746755">

Reproduction data:
```json
{
  "nodes": [
    {
      "name": "Income"
    },
    {
      "name": "Budget"
    },
    {
      "name": "Investment"
    },
    {
      "name": "Real Estate"
    },
    {
      "name": "Crypto"
    },
    {
      "name": "Stocks & Funds"
    },
    {
      "name": "Saving"
    },
    {
      "name": "Scpi"
    },
    {
      "name": "BTC"
    },
    {
      "name": "ETH"
    },
    {
      "name": "SOL"
    },
    {
      "name": "Housing"
    },
    {
      "name": "Food"
    },
    {
      "name": "Rent"
    },
    {
      "name": "Utility"
    },
    {
      "name": "Mortgage"
    },
    {
      "name": "Groceries"
    },
    {
      "name": "Delivery"
    },
    {
      "name": "Restaurant"
    }
  ],
  "links": [
    {
      "source": 0,
      "target": 1,
      "value": 8500
    },
    {
      "source": 1,
      "target": 2,
      "value": 2300
    },
    {
      "source": 1,
      "target": 3,
      "value": 400
    },
    {
      "source": 1,
      "target": 4,
      "value": 1250
    },
    {
      "source": 2,
      "target": 5,
      "value": 1800
    },
    {
      "source": 2,
      "target": 6,
      "value": 500
    },
    {
      "source": 3,
      "target": 7,
      "value": 400
    },
    {
      "source": 4,
      "target": 8,
      "value": 500
    },
    {
      "source": 4,
      "target": 9,
      "value": 500
    },
    {
      "source": 4,
      "target": 10,
      "value": 250
    },
    {
      "source": 1,
      "target": 11,
      "value": 3384
    },
    {
      "source": 1,
      "target": 12,
      "value": 800
    },
    {
      "source": 11,
      "target": 13,
      "value": 1234
    },
    {
      "source": 11,
      "target": 14,
      "value": 150
    },
    {
      "source": 11,
      "target": 15,
      "value": 2000
    },
    {
      "source": 12,
      "target": 16,
      "value": 450
    },
    {
      "source": 12,
      "target": 17,
      "value": 200
    },
    {
      "source": 12,
      "target": 18,
      "value": 150
    }
  ]
}
```

**Comment by ckifer** [comment]
@daformat feel free to open that PR. The changes look good to me and making this configurable makes plenty of sense

**Comment by ckifer** [comment]
Thanks for the contribution! Will be out in the next minor

**Comment by ckifer** [comment]
released in 2.8

**Comment by WookieeWalker** [comment]
Thank you for this @daformat, helped my project a lot
